File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

string convert(int cnt){
    
    string str="";
    while(cnt){ // 4
        str+=to_string(cnt%2);
        cnt/=2;
    }
    reverse(str.begin(),str.end());
    return str;
}

vector<int> solution(string s) {
    vector<int> answer;
    
    vector<int>v;
    
    
    
    int ans=0;
    int ans2=0;
    
    while(s!="1"){
    
        int cnt=0;
        
        for(int i=0;i<s.size();i++){
            if(s[i]=='0'){
                ans2++;
                continue;
            }
            cnt++;
        }
        s=convert(cnt);
        cout<<s<<" ";
        ans++;
    }
    
    
    
    
    
    
    v.push_back(ans);
    v.push_back(ans2);
    return v;
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        
        int cnt=0;
        int replace = 0;
        
        while(s.length()!=1){
            
            cnt++;
            
            int before = s.length();
            
            s = s.replaceAll("0","");
            
            replace += before - s.length();
            
            s = Integer.toString(s.length(),2);
        }
        
        int[] answer = {cnt,replace};
        
        return answer;
    }
}